scripts/scr_battle_roster/scr_battle_roster.gml

```diff
@@ -1,349 +1,536 @@
+    //???=obj_drop_select;
+    //???=obj_controller
+    //--------------------------------------------------------------------------------------------------------------------
+
+    // show_message("Container:"+string(_battle_loci)+", number:"+string(_loci_specific)+", planet?:"+string(_is_planet));
+
+    var stop, okay, sofar, man_limit_reached, man_size_count;
+    stop = 0;
+    okay = 0;
+    sofar = 0;
+    man_limit_reached = false;
+    man_size_count = 0;
+
+
+    // Formation here
+    obj_controller.bat_devastator_column = obj_controller.bat_deva_for[new_combat.formation_set];
+    obj_controller.bat_assault_column = obj_controller.bat_assa_for[new_combat.formation_set];
+    obj_controller.bat_tactical_column = obj_controller.bat_tact_for[new_combat.formation_set];
+    obj_controller.bat_veteran_column = obj_controller.bat_vete_for[new_combat.formation_set];
+    obj_controller.bat_hire_column = obj_controller.bat_hire_for[new_combat.formation_set];
+    obj_controller.bat_librarian_column = obj_controller.bat_libr_for[new_combat.formation_set];
+    obj_controller.bat_command_column = obj_controller.bat_comm_for[new_combat.formation_set];
+    obj_controller.bat_techmarine_column = obj_controller.bat_tech_for[new_combat.formation_set];
+    obj_controller.bat_terminator_column = obj_controller.bat_term_for[new_combat.formation_set];
+    obj_controller.bat_honor_column = obj_controller.bat_hono_for[new_combat.formation_set];
+    obj_controller.bat_dreadnought_column = obj_controller.bat_drea_for[new_combat.formation_set];
+    obj_controller.bat_rhino_column = obj_controller.bat_rhin_for[new_combat.formation_set];
+    obj_controller.bat_predator_column = obj_controller.bat_pred_for[new_combat.formation_set];
+    obj_controller.bat_landraider_column = obj_controller.bat_land_for[new_combat.formation_set];
+    obj_controller.bat_scout_column = obj_controller.bat_scou_for[new_combat.formation_set];
+
+    var company = -1;
+    var v = 0;
+    var meeting = false;
+
+    instance_activate_object(obj_pnunit);
+
+    //For each company and the HQ
+    repeat(11) {
+        if (man_limit_reached) {
+            break;
+        }
+        company++;
+        v = 0;
+		// HACK The below comment + while definition are commented out, since they seem to cause a CTD on battle start. We'll need to take a better look soon.
+        // //For each marine in that company, while unit exists (either marine name or vehicle role, vehicles have no names saved)
+        // //Marines and vehicles get added AT THE SAME TIME, (index [0][1] adds marine AND vehicle at index at the same time for loop x)
+        // //This is possible since array for saving vehicles and marines are separated
+        // //v<300 is an arbitrary number, probably linked to a company unit limit somewhere
+        // // while ((deploying_unit.name[company, v] != "" ||
+        // //         deploying_unit.veh_role[company, v] != "") && v < 300) {
+		repeat(300){
```

Comment:
the fix itself: reverted the loop definition to an earlier version. After the deployment we'll need to take a look to improve this.